core/src/main/java/io/questdb/cutlass/http/processors/JsonQueryProcessor.java

```diff
@@ -310,6 +311,7 @@ private static void syntaxError(
     private void compileQuery(JsonQueryProcessorState state) throws SqlException, PeerDisconnectedException, PeerIsSlowToReadException {
         final long nanos = nanosecondClock.getTicks();
         final CompiledQuery cc = compiler.compile(state.getQuery(), sqlExecutionContext);
+        compiler.getEngine().storeTelemetry(cc.getType());
```

Comment:
lets pass the call site of storeTelemetry() via sqlExecuitionContext rather than CairoEngine

Reply:
If I pass the executionContext this will be potentially have the a security context that do not allow writes (when read only), by using my own context I can use:

```
sqlExecutionContext.with(AllowAllCairoSecurityContext.INSTANCE, null, null);
```

Reply:
you can have the same security context as assigned to the execution context :)